### DISCUSSION TITLE
About rawValueWrapper

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#124] [General]
As you said, this side effect is experimental and can be removed at any time, okay, but... is not necessary to have a way to change a value in a not notified/rebuild way? I would like to have this possibility without fearing it can be removed.

Maybe there is that possibility and I'm ignorant of it

**Comment by GregoryConrad** [comment]
Once I figure out what to do with that `ValueWrapper` idea I mentioned in the other issue this would be resolved since that would allow you to have a value you can mutate in a build and outside a build (and I'd remove the `rawValueWrapper`).

**Reply by GregoryConrad** [reply]
I think I want to go down the keeping compatibility route, and just introduce a new `use.data` side effect or similar for this purpose. Then, I can figure out what I might want to do with use.state/use.stateGetterSetter, maybe just a deprecate.

**Comment by busslina** [comment]
But that mutation (inside a build) will trigger a rebuild? Because I think I need a way to mutate/change data without triggering a rebuild.
E.G. 

I don't want to rebuild on `setTimer`.

```dart
RamUsagePeriodicCheckerController _ramUsagePeriodicCheckerCapsule(
    CapsuleHandle use) {
  final (getTimer, setTimer) = use.rawValueWrapper<Timer?>(() => null);

  final systemModule = use(systemModuleCapsule);
  final alertsModule = use(alertsModuleCapsule);
  final log = use(loggerCapsule);

  return RamUsagePeriodicCheckerController(
    start: () {
      // Bad usage
      if (getTimer() != null) {
        throw ('RamUsagePeriodicCheckerController.start() -- Bad usage -- Already started');
      }

      setTimer(Timer.periodic(const Duration(seconds: 5), (_) async {
        final ramInfo = await OsUtils.getRamMemoryInfo();

        if (ramInfo == null) {
          log.error(systemModule.msg('Error obtaining RAM info'));
          return;
        }

        await alertsModule.alert(ramInfo.alert);
      }));
    },
    stop: () {
      // Bad usage
      if (getTimer() == null) {
        throw ('RamUsagePeriodicCheckerController.stop() -- Bad usage -- Already stopped');
      }

      getTimer()!.cancel();
      setTimer(null);
    },
  );
}
```

**Reply by GregoryConrad** [reply]
Ah. Setting the timer outside of the build would cause a rebuild in that case.

If it is an issue, you could then just toss the `RamUsagePeriodCheckerController` into a use.memo and then return that from the capsule. Even though the `_ramUsagePeriodicCheckerCapsule` would rebuild, the data would stay the same, so nothing downstream would be rebuilt.
